src/cocotb/share/lib/vpi/VpiCbHdl.cpp

```diff
@@ -239,14 +239,14 @@ int VpiSignalObjHdl::initialise(const std::string &name,
     } else {
         m_num_elems = vpi_get(vpiSize, GpiObjHdl::get_handle<vpiHandle>());
 
-        if (GpiObjHdl::get_type() == GPI_STRING) {
+        if (vpiConstant == type || vpiParameter == type ||
+            vpiStringVar == type) {
             m_indexable = false;  // Don't want to iterate over indices
             m_range_left = 0;
             m_range_right = m_num_elems - 1;
```

Comment:
This is kind of unfortunate. There's no way to get the bounds on the parameter typespec, so cocotb indexes and simulator indexes will be different. Maybe we can try to get the bounds from the typespec in the future.

Reply:
`parameter` objects have `vpiLeftRange` and `vpiRightRange` (which may be null if it has no explicitly defined range)

Reply:
I'm fine with getting this in now and filing an issue for a later fix of ranges.